### PR TITLE
Mailchimp Block: Replace submit button with inner button block

### LINKED ISSUE #15712: Mailchimp: Replace with SubmitButton new Button block
#### Is your feature request related to a problem? Please describe.

Now we have a [shared button block](https://github.com/Automattic/jetpack/pull/15370), we should update the Mailchimp block to use this instead of the SubmitButton component.

From https://github.com/Automattic/jetpack/issues/14761

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Fixes #15712

#### Changes proposed in this Pull Request:
* Replace the SubmitButton in the Mailchimp block with new shared Button block.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No.

#### Does this pull request change what data or activity we track or use?
* No.

#### Prerequisites:
* Mailchimp account
* Site with Mailchimp connection setup
* Post with Mailchimp block created prior to applying this PR's changes

#### Testing instructions:
1. Apply this PR
2. On the frontend, visit the post containing the previous version of Mailchimp block
3. Confirm that the block displays and functions as it did before
4. Edit the post
5. Confirm Mailchimp block displays and has correct settings
6. Update the Mailchimp block's button using new settings controls provided by button block
7. Save the post
8. Re-visit the post on the frontend and confirm display and function of block.

#### Before:
Screen Shot 2020-06-16 at 5 23 05 pm

#### After:
Screen Shot 2020-06-16 at 5 44 07 pm

#### Proposed changelog entry for your changes:
* Update Mailchimp block to use new shared Button block.

### CODE REVIEW

File: extensions/blocks/mailchimp/mailchimp.php [hunk 1]
```diff
@@ -114,9 +72,7 @@ function load_assets( $attr ) {
 					method="post"
 					id="mailchimp_form"
 					target="_top"
-					<?php if ( $is_amp_request ) : ?>
 					on="submit-success:AMP.setState( { mailing_list_status: 'subscribed', mailing_list_email: event.response.email } )"
-					<?php endif; ?>
```
Comment (Copons) [inline extensions/blocks/mailchimp/mailchimp.php:119]: Uh, nice catch here!
I was almost all like _HEY YOU! Why did you remove this?_ but gladly I double-checked first! 😄

File: extensions/blocks/mailchimp/mailchimp.php [hunk 2]
```diff
@@ -150,11 +106,7 @@ class="mc-submit-param"
 						value="<?php echo esc_attr( $values['signupFieldValue'] ); ?>"
 					/>
 				<?php endif; ?>
-				<p>
-					<button type="submit" class="<?php echo esc_attr( $button_classes ); ?>" style="<?php echo esc_attr( $button_styles ); ?>">
-						<?php echo wp_kses_post( $values['submitButtonText'] ); ?>
-					</button>
-				</p>
+				<?php echo render_button( $attr, $content ); // phpcs:ignore ?>
```
Comment (Copons) [inline extensions/blocks/mailchimp/mailchimp.php:109]: I'll leave it to JP folks to chime in, but I wonder if it would make more sense to only disable the broken PHPCS rule:
```php
// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
echo render_button( $attr, $content );
// phpcs:enable WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment (fgiannar) [reply extensions/blocks/mailchimp/mailchimp.php:109]: I'd agree with that as it gives as a clear look at why and what we are ignoring 👍

**Comment by fgiannar** [comment]
Other than the PHPCS rule comment above, everything looks good, thanks!

File: extensions/blocks/mailchimp/mailchimp.php (hunk 2 above)
Comment (jeherve) [reply extensions/blocks/mailchimp/mailchimp.php:109]: I believe you could even do it on one line:

```suggestion
				<?php echo render_button( $attr, $content ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
```

**Comment by jeherve** [comment]
r210103-wpcom
